hibernate-core/src/main/java/org/hibernate/query/criteria/internal/expression/function/ParameterizedFunctionExpression.java

```diff
@@ -111,7 +111,8 @@ public String render(RenderingContext renderingContext) {
 	protected void renderArguments(StringBuilder buffer, RenderingContext renderingContext) {
 		String sep = "";
 		for ( Expression argument : argumentExpressions ) {
-			buffer.append( sep ).append( ( (Renderable) argument ).render( renderingContext ) );
+			// invoking renderProjection to avoid the binding of the returning literals
+			buffer.append( sep ).append( ( (Renderable) argument ).renderProjection( renderingContext ) );
```

Comment:
What is the rationale leading you to this fix?

The `LiteralHandlingMode` documentation says `The BIND mode will use bind variables for any literal value.`. So the previous behavior of rendering them as parameters seems legit to me and the intended behavior.

IMHO, the issue here is that the type of the parameter cannot be determined as it's lost in the JPQL translation.

I'll create a thread on the list to discuss my findings.